lib/puppet/provider/user/useradd.rb

```diff
@@ -152,6 +152,46 @@ def groups=(value)
     set(:groups, value)
   end
 
+  def password=(value)
+    user = @resource[:name]
+
+    begin
+      # Puppet execute does not support strings as input, only files.
+      # The password is expected to be in an encrypted format given -e is specified:
+      tempfile = nil
+      tempfile = Tempfile.new("puppet_#{user}_pw", :encoding => Encoding::ASCII)
```

Comment:
Suggest not leaking the user's name here or including `_pw`. Instead maybe just prefix as `puppet`?